File: 프로그래머스/unrated/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    var answer = 0;
    const idx = Math.max(...array)
    const set = new Set(array)
    const arr = Array.from(set)
    const total = new Array(idx+1).fill(0)
    
    arr.forEach((item) => {
        let cnt = 0
        array.forEach((tem) => {
            if (tem === item) {
                cnt += 1
            }
        })
        total[item] = cnt
    })
    
    
    const mx = Math.max(...total)
    
    
    const gap = total.filter((item) => item === mx)
    console.log(gap)
    
    if (gap.length > 1) {
        answer = -1
    } else {
        answer = total.indexOf(mx)
    }
    
    

    return answer;
}

File: 프로그래머스/unrated/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    idx = max(array)
    setarr = set(array)
    arr = list(setarr)
    total = [0] * (idx + 1)
    
    answer = 0
    for i in arr:
        total[i] = array.count(i)
    
    mx = max(total)
    ii = total.count(mx)
    if ii > 1:
        answer = -1
    else:
        answer = total.index(mx)

    return answer